docs/tutorials/time-series.md

```diff
@@ -0,0 +1,766 @@
+of the input rows by a specified amount of time.  More precisely, given a `WATERMARK` annotation with value `WM`,
+an input row with a value `X` for the watermarked column will be "held up" until
+the stream time of the table reaches `X + WM`, at which point the program will
+behave as if the row with value `X` has only just been received.
+
+This delay allows `WM` time units for out-of-order data to arrive.
+For a column with a `LATENESS` annotation, setting `WATERMARK` to be equal
+to `LATENESS` ensures that all received data is processed in-order.
+
+`WATERMARK` is specified as an expression that evaluates to a constant value.
+The expression must have a type that can be subtracted from the column
+type.  For example, a column of type `TIMESTAMP` may have a watermark
+specified as an `INTERVAL` type:
+
+```sql
+CREATE TABLE purchase_watermark (
+   customer_id INT,
+   ts TIMESTAMP NOT NULL WATERMARK INTERVAL 1 HOURS,
+   amount BIGINT
+);
+```
+
+:::warning
+
+The current version of the SQL compiler does not support multiple
+`WATERMARK` columns in a single table.
+
+:::
+
+## Append-only tables
+
+Time series tables are often append-only: once an event has been received, it
+cannot be modified or deleted.  Feldera is able to leverage this property to
+optimize certain types of queries. Consider the `daily_max` view from above:
+
+```sql
+CREATE VIEW daily_max AS
+SELECT
+    TIMESTAMP_TRUNC(ts, DAY) as d,
+    MAX(amount) AS max_amount
+FROM
+    purchase
+GROUP BY
+    TIMESTAMP_TRUNC(ts, DAY);
+```
+
+Maintaining the `MAX` aggregate incrementally requires permanently storing the
+entire contents of the `purchase` table (ignoring the `LATENESS` annotation on
+`purchase.ts`).  However, if the table is append-only, then the `MAX` aggregate
+can only increase monotonically as new records are added to the table, and we
+only need to store the current largest value for each 1-day window.
+
+The `append_only` annotation on a table instructs Feldera that the table will
+only receive `INSERT` updates, enabling this and other optimizations:
+
+```sql
+CREATE TABLE purchase (
+   customer_id INT,
+   ts TIMESTAMP NOT NULL LATENESS INTERVAL 1 HOURS,
+   amount BIGINT
+) WITH (
+    'append_only' = 'true'
+);
+```
+
+Note that `append_only` annotations are only supported for tables; however
+the Feldera query engine automatically derives it for views that depend on
+append-only tables.
+
+## SQL for time series analytics
+
+In this section, we present a catalog of SQL operators and patterns frequently used in
+time series analysis. Although these operators are applicable to both time-series and
+non-time-series data, they are particularly beneficial when working with time series.
+
+Feldera offers efficient incremental implementations of these constructs and
+supports garbage collection for most of them, meaning that they can be evaluated
+efficiently in both time and space.
+
+### Aggregating Time Series Data Over Fixed-Size Time Intervals
```

Comment:
This section is more important that the algorithms and deep dives so far. I'd consider reordering.

Reply:
I added a reference to it in the preamble.